streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -3444,25 +3444,24 @@ object ZStream extends ZStreamPlatformSpecificConstructors {
   def fromIterableM[R, E, O](iterable: ZIO[R, E, Iterable[O]]): ZStream[R, E, O] =
     fromEffect(iterable).mapConcat(identity)
 
-  def fromIterator[A](iterator: => Iterator[A]): ZStream[Any, Throwable, A] = {
+  def fromIterator[A](
+    iterator: => Iterator[A],
+    chunkSize: Int = ZStream.DefaultChunkSize
+  ): ZStream[Any, Throwable, A] = {
     object StreamEnd extends Throwable
-
-    ZStream.fromEffect(Task(iterator) <*> ZIO.runtime[Any]).flatMap {
+    ZStream.fromEffect(Task(iterator.grouped(chunkSize)) <*> ZIO.runtime[Any]).flatMap {
```

Comment:
Wouldn't it be more efficient to avoid constructing the intermediate `List` from Iterator.grouped() and instead manually `take()` one by one, appending to a `Chunk` until `hasNext == false` or `chunkSize` is reached?

Reply:
Although this is essentially running the initial implementation through a `ZTransducer.collectAllN`. isn't it?  Should the new impl simply be `original >>> ZTransducer.collectAllN(chunkSize)`?